sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -412,7 +412,8 @@ object CatalystTypeConverters {
     case d: JavaBigDecimal => new DecimalConverter(DecimalType(d.precision, d.scale)).toCatalyst(d)
     case seq: Seq[Any] => new GenericArrayData(seq.map(convertToCatalyst).toArray)
     case r: Row => InternalRow(r.toSeq.map(convertToCatalyst): _*)
-    case arr: Array[Any] => new GenericArrayData(arr.map(convertToCatalyst))
+    case arr: Array[Byte] => arr
```

Comment:
Actually it is tricky as `Array[Byte]` could be `ArrayType` of `ByteType` and `BinaryType` either. 

In `convertToCatalyst` and `Literal.apply` where we convert scala objects to catalyst types, we'd take `Array[Byte]` as `BinaryType`.